algorithms/prims-algorithm/prims-algorithm.tex

```diff
@@ -0,0 +1,63 @@
+\section{Prim's Algorithm}\index{graph!Prim's Algorithm}
+#ifdef hackpackpp
+Prims algorithm is a greedy algorithm that allows for finding a minimum spanning tree/forest in $O(E + \log{V})$ time.
+Big picture, Prims algorithm works by doing the following:
+
+\begin{enumerate}
+	\item Choose an arbitrary unvisited node in the graph.
+	\item Grow the tree by choosing the node that can be added to the tree by least cost.
+	\item Repeat step 2 until all connected nodes have been visited. If there are still nodes, return to step 1.
+\end{enumerate}
+
+#endif
+\subsection{Applications}
+\begin{itemize}
+	\item Finding a minimum spanning tree
+\end{itemize}
+
+#ifdef hackpackpp
+\subsection{Example Contest Problem: Cow Connection}
+Farmer John has not been very diligent in mowing his farm.
+As a result, the grazing field now resembles a maze with small openings at the intersections.
```

Comment:
I don't understand what "small openings at the intersections" means. Small clearings? Is it important to mention this?